### PR TITLE
[BarcodeScanner][iOS] Fix scanning PDF417 with \0 (null) characters

### LINKED ISSUE #4817: BarcodeScanner truncates data if QR code in PDF417 format contains null characters in iOS
### Environment

Expo CLI 2.19.4 environment info:
    System:
      OS: macOS 10.14.6
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 10.15.0 - /usr/local/bin/node
      npm: 6.9.0 - ~/.npm-packages/bin/npm
    IDEs:
      Android Studio: 3.1 AI-173.4697961
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.1 => 32.0.6 
      react: 16.3.1 => 16.3.1 
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1 
      react-navigation: ^3.11.0 => 3.11.0 
    npmGlobalPackages:
      expo-cli: 2.19.4

iOS (works fine in Android)

### Steps to Reproduce

1. Scan a QR code in PDF417 format (this might extend beyond this format), that has a non readable sequence encoded in it, I'm scanning a barcode that begins with:
`40 0a 1e 0d 41 4e 53 49 20 39 39 30 38 37 36 30 33 30 30 30 31 44 4c 1f 00 00 00 c3 85 00 00 00 44 4c 44 43 ...`
This belongs to an Ontario Driver License, and my guess is that the offending sequence is: `1f 00 00 00 c3 85 00 00 00` (a Unit Separator Character, followed by three nulls, then a Next Line, then three nulls). I've created a QR linked at the end of the description that you can scan in Android and iOS and it shows how the character `\0x00` truncates the output in iOS.

2. The iOS object returned by the `onBarCodeScanned` callback has the `data` attributed truncated when that sequence begins. That means I can only read the first few characters of that Barcode. The behavior in Android works well, it reads the entire barcode.

### Expected Behavior

The entire content of the QR code is returned in the data field.

### Actual Behavior

Only the characters before the offending sequence are returned.

### Reproducible Demo

```javascript
import React, { Component } from 'react';
import {StyleSheet, Text, View} from 'react-native';

import {Permissions} from 'expo-permissions';

import {BarCodeScanner} from 'expo-barcode-scanner';
import {SplashScreen} from "expo";

export default class App extends Component {
  constructor(props) {
    super(props);
  }

  state = {
    hasCameraPermission: null
  };

  async componentDidMount() {
    SplashScreen.hide();
    this.getPermissionsAsync();
  }

  getPermissionsAsync = async () => {
    const {status} = await Permissions.askAsync(Permissions.CAMERA);
    this.setState({hasCameraPermission: status === 'granted'});
  };

  render() {
    const {hasCameraPermission} = this.state;

    if (hasCameraPermission === null) {
      return <Text>Requesting for camera permission</Text>;
    }
    if (hasCameraPermission === false) {
      return <Text>No access to camera</Text>;
    }
    return (
        <View
            style={{
              flex: 1,
              flexDirection: 'column',
              justifyContent: 'flex-end',
            }}>

          <BarCodeScanner
              onBarCodeScanned={this.handleBarCodeScanned}
              style={StyleSheet.absoluteFillObject}
          />
        </View>
    );
  }

  handleBarCodeScanned = (barcode) => {
    console.log(barcode.data)
  }
}
```
[QR code to test](https://imgur.com/c2rEZ1V)

This was already reported: https://forums.expo.io/t/null-characters-are-breaking-the-data-when-using-barcode-scanner-on-ios/10923

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

Resolves #4817
iOS SDK does not properly scan `PDF417` barcodes containing `u'\0'` (null) characters, what trims substring after `u'\0'` character occurrence.

_Update:_
Added support for scanning `Code39` by `ZXing` as it scans correctly images [like 
this](https://github.com/expo/expo/pull/5976#issuecomment-545001008) while built-in iOS scanner does not recognise barcode at all.

# How

I've added `zxing` library for scanning only `PDF417` barcode type.

# Test Plan

`ncl#BarcodeScannerScreen`

### CODE REVIEW

**Review by tsapeta — CHANGES_REQUESTED** [review]
This whole ZXing pod looks really big, can we use just `PDF417` subspec instead?

**Comment by xiCO2k** [comment]
Hey @bbarthec also detected that code39 is not working well for example with car VIN codes like this:
image

Do you think the ZXing library can also help with that?

Thanks

**Comment by xiCO2k** [comment]
@bbarthec thats awesome, thanks ;)

**Comment by bbarthec** [comment]
@xiCO2k, it looks that `ZXing` can handle your scenario and produce some useful result 🎉 I've added support for scanning `Code39` for ZXing flow. Meanwhile you can try [`ZXing demo app`](https://github.com/zxingify/zxingify-objc/tree/master/examples/BarcodeScanner).

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ This is the log of notable changes to the Expo client that are developer-facing.
 - Fixed `Calendar.getCalendarsAsync` rejecting on iOS 13, when `source.name` is null. ([#5994](https://github.com/expo/expo/pull/5994) by [@lukmccall](https://github.com/lukmccall))
 - Fixed handling URI with no scheme in `ExpoFileSystem`. ([#5904](https://github.com/expo/expo/pull/5904) by [@bbarthec](https://github.com/bbarthec))
 - Fixed `FileSystem#deleteAsync` in older Android SDKs. ([#5923](https://github.com/expo/expo/pull/5923) by [@bbarthec](https://github.com/bbarthec))
+- Fixed scanning `PDF417` in `BarCodeScanner` on iOS. ([#5976](https://github.com/expo/expo/pull/5531) by [@bbarthec](https://github.com/bbarthec))
```
Comment (tsapeta) [inline CHANGELOG.md]:
```suggestion
- Fixed scanning `PDF417` and `Code39` in `BarCodeScanner` on iOS. ([#5976](https://github.com/expo/expo/pull/5531) by [@bbarthec](https://github.com/bbarthec))
```

**Review by tsapeta — APPROVED** [review]

**Comment by xiCO2k** [comment]
@bbarthec @tsapeta That is great, thanks a lot.

**Review by esamelson — COMMENTED** [review]
thanks for looking into this @bbarthec . Code changes look good to me, but could we get a sense of the filesize diff of a release-mode build with and without these three ZXing pods included?

**Comment by gabzim** [comment]
Will this make SDK 36?

**Comment by xiCO2k** [comment]
Hopping that will be on the SDK 36, the Code39 fix for iOS would be awesome to have it no SDK 36 :)

**Comment by bbarthec** [comment]
@esamelson, I've prepared two simulator builds (based off master & based off this branch) using `et client-build -p ios` command and these are the results:
```bash
150M	ExponentWithoutZXING.app
152M	ExponentWithZXING.app
231M	ExponentWithoutZXING.app.dSYM
233M	ExponentWithZXING.app.dSYM
```

**Review by esamelson — APPROVED** [review]
OK, that diff doesn't seem too bad, let's go ahead and land this. Thanks for investigating!

**Pull request merged**